### DISCUSSION TITLE
How to get user.id from event_join?

### DISCUSSION THREAD

**Discussion by MasterRelaxa** [TwitchIO/TwitchIO#309] [Questions] [answered]
Hello,

How could i get the `user.id` from `event_join`, if it's always set to `None`?

as example:
```
async def event_join(self,channel,user):
print(f"{user.id,user.name}")

(None, 'icewizerds') 
(None, 'alienconglomeration')
```
Also tried to get it from event_ready and event_channel_joined and had the same issue.

I need to get a unique user-id, to be able to stack points in case the user changed the name.

**Comment by chillymosh** [comment]
This data is not available via IRC join messages directly, as you can see from the raw messages
```
:nightbot!nightbot@nightbot.tmi.twitch.tv JOIN #chillymosh
:streamelements!streamelements@streamelements.tmi.twitch.tv JOIN #chillymosh
```

You'd have to make an API call to do so.

**Comment by MasterRelaxa** [comment, marked as answer]
It can be fetched.
```
    async def event_join(self,channel,user):
        userdata = await self.fetch_users([user.name])
        print(userdata[0].id)
```
Thanks for the help in discord.

**Comment by chillymosh** [comment]
Note for others to be aware that event_join can and will be called multiple times for same user even if they haven't left and rejoined due to how Twitch batch send.
